src/ptf/testutils.py

```diff
@@ -3453,6 +3453,70 @@ def verify_any_packet_any_port(
     return match_index
 
 
+def verify_any_masked_packet_any_port(
+    test, pkts=[], ports=[], device_number=0, timeout=None, n_timeout=None
+):
+    """
+    a.) Check that _any_ of the packets is received on _any_ of the specified ports belonging to
+    the given device (default device_number is 0).
+
+    b.) Also verifies that the packet is not received on any other ports for this
+    device, and that no other packets are received on the device (unless --relax
+    is in effect).
+
+    In contrast to standard verify_any_packet_any_port() function this one
+    allows to verify masked packets.
+
+    Returns the index of the port on which the packet is received.
+
+    Timeout for this function is used as +ve timeout for (a) and -ve timeout for (b)
+    Note: +ve timeout here means timeout in which we are expecting pkt to arrive in
+    -ve timeout here means timeout for which we will wait for to check for unexpected pkts
+    """
+    if timeout == None:
+        timeout = ptf.ptfutils.default_timeout
+    if n_timeout == None:
+        n_timeout = ptf.ptfutils.default_negative_timeout
+
+    if timeout <= 0 or n_timeout <= 0:
+        raise Exception(
+            "%s() requires positive timeout value." % sys._getframe().f_code.co_name
+        )
+
+    received = False
+    match_index = 0
+
+    result = dp_poll(test, device_number=device_number, timeout=timeout)
+
+    if isinstance(result, test.dataplane.PollSuccess) and result.port in ports:
+        received_packet = str(result.packet)
+        for pkt in pkts:
+            if ptf.dataplane.match_exp_pkt(pkt, received_packet):
```

Comment:
but what if the exact same packet will be received at two ports? Do we only save port id with higher index values?

Reply:
We rather expect here that any packet is received once on one of given ports (example use case is LAG ports verification)

Reply:
Maybe that's a solution but I think we shouldn't change the type of a value returned by this function. We should keep backward compatibility.

Reply:
I think this is the correct fix to make though. Every other instance of a verify packet in this file uses match_exp_pkt via dp_poll API rather than using ==. This test is an aberration and this change addresses it.